src/System.Private.ServiceModel/src/System/ServiceModel/Description/XmlSerializerOperationBehavior.cs

```diff
@@ -412,6 +412,12 @@ private void CreateHeaderDescriptionTable(MessageDescription message, MessageInf
                         else if (headersMapping != null)
                         {
                             XmlMemberMapping memberMapping = headersMapping[headerNameIndex++];
+#if FEATURE_NETNATIVE
```

Comment:
Unfortunately it's kind of clunky how to do this. If you look at the [source](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs) for RuntimeInformation, you'll see that the string FrameworkDescription is constructed like this:
```csharp
return string.Format("{0} {1}", FrameworkName, typeof(object).GetTypeInfo().Assembly.GetName().Version);
```
Where FrameworkName is conditionally compiled to be the constant value `".NET Native"`. This means we need the following code:
```csharp
    bool isNativeCompiled = RuntimeInformation.FrameworkName.StartsWith(".NET Native");
```
This is a relatively expensive string operation to do each time. Can you add a static helper method to a (hopefully pre-existing) class in our code which caches the result of this so it's a simple bool check. I'm adding a dependency on RuntimeInformation in another PR that I'm just about to merge in the next hour or so.